File: beakjoon/15000-16000/15650/15650.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>

using namespace std;

int N, M;
int arr[] = {
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
};

void print_set(vector<int> &selected, int prev, int toPick)
{
    if (toPick == 0)
    {
        for (int i = 0; i < selected.size(); i++)
            cout << selected[i] << " ";
        cout << "\n";
        return;
    }

    for (int i = prev + 1; i < N; i++)
    {
        selected.push_back(arr[i]);
        print_set(selected, i, toPick - 1);
        selected.pop_back();
    }
}
int main()
{
    cin >> N >> M;
    vector<int> v;

    print_set(v, -1, M);

    return 0;
}

File: BOJ/15000-16000/15650/15650.py
N, M = map(int, input().split())


def print_set(arr, selected, prev, toPick):

    if toPick == 0:
        print(*selected)
        return

    for i in range(prev+1, len(arr)):
        print_set(arr, selected+[arr[i]], i, toPick-1)


print_set([i+1 for i in range(N)], [], -1, M)
